libs/libc/pwd/lib_pwd.h

```diff
@@ -37,12 +38,14 @@
 #  define CONFIG_LIBC_PASSWD_LINESIZE 80
 #endif
 
-#define ROOT_NAME  "root"
-#define ROOT_UID   0
-#define ROOT_GID   0
-#define ROOT_GEOCS "root"
-#define ROOT_DIR   "/root"
-#define ROOT_SHELL "/bin/nsh"
+#define ROOT_NAME   "root"
+#define ROOT_UID    0
+#define ROOT_GID    0
+#define ROOT_GEOCS  "root"
+#define ROOT_DIR    "/root"
+#define ROOT_SHELL  "/bin/nsh"
+#define ROOT_PASSWD "root"
+#define ROOT_PWDP   "$1$123$SGj4CnC7VtiFx.tjjtazK1"
```

Comment:
Please don't store fixed/hard-coded password, get it from /etc/passwd search how it is done currently and compared against user configured CONFIG_NSH_LOGIN_PASSWORD